air-quality-backend/src/air_quality/etl/forecast/forecast_adapter.py

```diff
@@ -107,3 +113,309 @@ def transform(forecast_data: ForecastData, locations: list[AirQualityLocation])
+        format = "L"
+
+    image = Image.fromarray(image_array, format)
+    output_file = (
+        f"{output_directory}/{variable}_{forecast_date}_CAMS_global."
+        f"chunk_{num_chunk}_of_{total_chunks}.{file_format}"
+    )
+    image.save(output_file, format=file_format, lossless=True)
+    return output_file
+
+
+def _process_variable(
+    data: xr.Dataset,
+    data_type: str,
+    variable_name: str,
+    output_directory: str,
+    forecast_date: str,
+) -> List[Dict[str, str]]:
+    """
+    Process a specific variable and create data textures.
+
+    :param data: The dataset containing the variable.
+    :param data_type: The type of the data.
+    :param variable_name: The name of the variable.
+    :param output_directory: Directory to save the textures.
+    :param forecast_date: The forecast date.
+
+    :return: A list of dictionaries containing data texture metadata for database.
+    """
+    documents = []
+    chunks_per_texture = 16
+    file_format = "webp"
+
+    rgb_data_array, min_value, max_value, units, num_lon, time_vector = _convert_data(
+        data, data_type
+    )
+    chunk_list, chunk_dict = _chunk_data_array(
+        rgb_data_array, chunks_per_texture, num_lon, time_vector
+    )
+
+    # WebP has better compression, but maximum pixel dimension is 16383 x 16383.
+    if (num_lon * chunks_per_texture) > 16383:
+        file_format = "png"
+
+    for num_chunk, chunk in enumerate(chunk_list):
+        output_file = _save_data_texture(
+            chunk,
+            output_directory,
+            forecast_date,
+            variable_name,
+            num_chunk + 1,
+            len(chunk_list),
+            file_format,
+        )
+        document = {
+            "forecast_base_time": forecast_date,
+            "variable": variable_name,
+            "source": "cams-production",
+            "min_value": min_value,
+            "max_value": max_value,
+            "units": units,
+            "texture_uri": output_file,
+            "time_start": chunk_dict[num_chunk]["time_start"],
+            "time_end": chunk_dict[num_chunk]["time_end"],
+            "chunk": f"{num_chunk+1} of {len(chunk_list)}",
+        }
+        documents.append(document)
+    return documents
+
+
+def create_data_textures(forecast_data: ForecastData):
+    """
+    Create gridded data textures from forecast data for frontend maps
+    :param forecast_data:
+    :return: List of dictionaries containing data texture metadata for database
+    """
+    output_directory, forecast_date = _create_output_directory(forecast_data)
+    db_metadata = []
+
+    for pollutant in PollutantType:
+        logging.info(f"Creating data textures for {pollutant.name}")
+        forecast_data_type = convert_to_forecast_data_type(pollutant)
+        dataset = forecast_data._get_data_set(forecast_data_type)
+        documents = _process_variable(
+            dataset,
+            forecast_data_type.value,
```

Comment:
I think these are basically equivalent, are we able to just use one or the other (I think they get passed around a fair bit so might need to change other functions too)?

Reply:
I simplified this a bit